inference-engine/src/mkldnn_plugin/nodes/psroi.cpp

```diff
@@ -42,250 +46,418 @@ class PSROIPoolingImpl: public ExtLayerBase {
+
+                    const float subBinSizeH = binSizeH / static_cast<float>(spatialBinsX);
+                    const float subBinSizeW = binSizeW / static_cast<float>(spatialBinsY);
+
+                    const int partH = h * partSize / pooledHeight;
+                    const int partW = w * partSize / pooledWidth;
+                    const int classId = c / channelsEachClass;
+                    const float transX = noTrans ? 0 :
+                                         bottomTrans[(((n * numClasses + classId) * 2) * partSize + partH)
+                                                     * partSize + partW] * transStd;
+                    const float transY = noTrans ? 0 :
+                                         bottomTrans[(((n * numClasses + classId) * 2 + 1) * partSize + partH)
+                                                     * partSize + partW] * transStd;
+
+                    const float wStart = w * binSizeW + roiStartW + transX * roiWidth;
+                    const float hStart = h * binSizeH + roiStartH + transY * roiHeight;
+
+                    float sum = 0;
+                    int count = 0;
+                    int gw = w * groupSize / pooledWidth;
+                    int gh = h * groupSize / pooledHeight;
+                    gw = (std::min)((std::max)(gw, 0), static_cast<int>(groupSize - 1));
+                    gh = (std::min)((std::max)(gh, 0), static_cast<int>(groupSize - 1));
+
+                    const inputType *offsetBottomData = srcData + (roiBatchInd * channels) * height * width;
+                    for (size_t ih = 0; ih < spatialBinsY; ih++) {
+                        for (size_t iw = 0; iw < spatialBinsX; iw++) {
+                            float w1 = wStart + iw * subBinSizeW;
+                            float h1 = hStart + ih * subBinSizeH;
+                            // bilinear interpolation
+                            if (w1 < -0.5 || w1 > width - 0.5 || h1 < -0.5 || h1 > height - 0.5)
+                                continue;
+                            w1 = static_cast<float>((std::min)((std::max)(static_cast<double>(w1), 0.0), width - 1.0));
+                            h1 = static_cast<float>((std::min)((std::max)(static_cast<double>(h1), 0.0), height - 1.0));
+                            int c1 = static_cast<int>((c * groupSize + gh) * groupSize + gw);
+                            float val = bilinearInterp<inputType>(offsetBottomData +
+                                                                  c1 * height * width, w1, h1, width,
+                                                                  inBlockSize, c1 % inBlockSize);
+
+                            sum += val;
+                            count++;
+                        }
+                    }
+                    dstData[dstIndex] = count == 0 ? 0 : sum / count;
+                });
             }
         });
 
-        for (int n = real_rois; n < nn; n++) {
+        for (int n = realRois; n < nn; n++) {
```

Comment:
Isn't it better to use memset here?

Reply:
We don't care about all rois, after first one with -1 batch index

Reply:
Done